File: apps/notebook/src/lib/frame-types.ts
/**
 * Frame type constants matching `notebook_doc::frame_types` in Rust.
 *
 * These correspond to the first byte of each typed frame payload on
 * notebook sync connections. Defined here so the frontend can reference
 * them without magic numbers.
 *
 * Outbound frames flow through `host.transport.sendFrame(frameType, payload)`
 * — there's no longer a top-level `sendFrame()` helper that reaches for
 * Tauri directly.
 */
export const frame_types = {
  /** Automerge sync message (binary). */
  AUTOMERGE_SYNC: 0x00,
  /** NotebookRequest (JSON). */
  REQUEST: 0x01,
  /** NotebookResponse (JSON). */
  RESPONSE: 0x02,
  /** NotebookBroadcast (JSON). */
  BROADCAST: 0x03,
  /** Presence (CBOR, see notebook_doc::presence). */
  PRESENCE: 0x04,
  /** RuntimeStateSync message (binary Automerge sync for RuntimeStateDoc). */
  RUNTIME_STATE_SYNC: 0x05,
  /** PoolStateSync message (binary Automerge sync for PoolDoc, global). */
  POOL_STATE_SYNC: 0x06,
} as const;

File: crates/notebook-doc/src/frame_types.rs
//! Frame type constants shared across all consumers (daemon, WASM, Python).
//!
//! These correspond to the first byte of each typed frame payload on
//! notebook sync connections. Defined here in `notebook-doc` so that
//! `runtimed` (daemon), `runtimed-wasm` (frontend), and `runtimed-py`
//! (Python bindings) all share one source of truth.
//!
//! The canonical `NotebookFrameType` enum with `TryFrom<u8>` lives in
//! `runtimed::connection` — these constants exist for consumers that
//! can't depend on `runtimed` (e.g., WASM).

/// Automerge sync message (binary).
pub const AUTOMERGE_SYNC: u8 = 0x00;

/// NotebookRequest (JSON).
pub const REQUEST: u8 = 0x01;

/// NotebookResponse (JSON).
pub const RESPONSE: u8 = 0x02;

/// NotebookBroadcast (JSON).
pub const BROADCAST: u8 = 0x03;

/// Presence (CBOR, see `presence` module).
pub const PRESENCE: u8 = 0x04;

/// RuntimeStateDoc sync message (binary Automerge sync).
/// Carries the per-notebook ephemeral runtime state (kernel status, queue, env sync).
/// Daemon-authoritative — client changes are stripped on receive.
pub const RUNTIME_STATE_SYNC: u8 = 0x05;
